### ISSUE TITLE
Reopening the buffer causes errors that have already been fixed to re-occur

### ISSUE THREAD

**Issue by suhdonghwi** [haskell/haskell-ide-engine#1629]
# Steps to reproduce

1. Make any kind of error
```haskell
doi -- << Error here 
  putStrLn "Hello, world!"
  return 1
```
2. Fix it
```haskell
do -- << Error has been fixed
  putStrLn "Hello, world!"
  return 1
```
3. Close and reopen the source file, and edit anywhere
```haskell
do
  putStrLn "Hello, world!"
  return $ 1 + 1 -- << Edited here
```
4. Error which occured in Step 1 occurs again at the same location
```haskell
do -- << Error here!
  putStrLn "Hello, world!"
  return $ 1 + 1
```

---

This is my output of `hie-wrapper --version`:
```
Version 1.0.0.0, Git revision 4bec07d79cf77c8ee855957d20e315a89f799d75 (3745 commits) x86_64 ghc-8.6.5
```

**Comment by fendor** [comment]
Is there auto-save or something? Or is it unsaved?

**Comment by suhdonghwi** [comment]
I manually saved the file before closing and reopening it.

**Comment by fendor** [comment]
Can not reproduce with vscode on nixos with hie-8.6.5 on a cabal project.
Can you add more information? What is the `hie` version, which one is used, what type of project, operating system, lsp-client and are you sure there are no other plugins active?

**Comment by suhdonghwi** [comment]
This is the output of `hie --version`:

`Version 1.0.0.0, Git revision 4bec07d79cf77c8ee855957d20e315a89f799d75 (3745 commits) x86_64 ghc-8.6.5`

I run it in Windows Subsystem for Linux, Ubuntu 18.04 LTS. Language server client is VSCode (WSL remote development) Haskell Language server extension. (But this problem also happens in neovim coc.nvim)

**Comment by fendor** [comment]
Can you provide the logs? If possible with `--debug` enabled. Either in vscode with `File > Preferences > Settings > Haskell Language Server > Language Server Haskell > Trace: Server` set to `verbose` or by invoking the hie/hie-wrapper process with the arguments `hie --lsp --debug`/`hie-wrapper --lsp --debug`

**Comment by suhdonghwi** [comment]
This is my log reproducing the bug: https://pastebin.com/ZRYwWn1H

**Comment by fendor** [comment]
And at the end errors were shown in the file `Expr.hs`?

**Comment by suhdonghwi** [comment]
Yes. But after reopening, if I edit anywhere and save the file again, errors disappear.

**Comment by fendor** [comment]
I can reproduce this error on windows as well, not reliably, but encountered it several times now. I suspect some caching issue, but I did not dig into it, yet.
